cpp/src/strings/regex/regcomp.cpp

```diff
@@ -246,11 +247,50 @@ class regex_parser {
     return {false, c};
   }
 
+  // for \d and \D
+  void add_ascii_digit_class(std::vector<reclass_range>& ranges, bool negated = false)
+  {
+    if (!negated) {
+      ranges.push_back({'0', '9'});
+    } else {
+      ranges.push_back({0, '0' - 1});
+      ranges.push_back({'9' + 1, std::numeric_limits<char32_t>::max()});
```

Comment:
Would it be worth defining a named constant `using MAX_REGEX_CHAR = std::numeric_limits<char32_t>::max()`? I'm specifically wondering because I don't know where (if anywhere) it's documented why we use `char32_t`. I see [this function](https://github.com/rapidsai/cudf/blob/branch-22.10/cpp/src/strings/regex/regcomp.cpp#L81) that converts from input UTF-8 to `char32_t`, but I'm not sure why our code is designed that way (performance for 4 byte reads maybe?).

Reply:
Introducing the constant makes sense.
The reason this is 4 bytes (`char32_t`) is because it represents a single UTF-8 character which can be 1-4 bytes.